cpp/src/io/parquet/rle_stream.cuh

```diff
@@ -0,0 +1,348 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "parquet_gpu.hpp"
+#include <cudf/detail/utilities/integer_utils.hpp>
+
+namespace cudf {
+namespace io {
+namespace parquet {
+namespace gpu {
+
+// TODO: consider if these should be template parameters to rle_stream
+constexpr int num_rle_stream_decode_threads = 512;
+constexpr int num_rle_stream_decode_warps   = (num_rle_stream_decode_threads / 32) - 1;
```

Comment:
why -1?

Reply:
There's 16 warps in the block.  1 warp is doing the read-ahead setup of runs while the remaining 15 are decoding the previous batch.